packages/backend/src/middlewares/requireRole.ts

```
@@ -0,0 +1,16 @@
+import { NextFunction, Response } from 'express';
+import { AuthenticatedRequest } from './requireAuth';
+
+/**
+ * Middleware to restrict access by user role
+ * @param allowedRoles Array of allowed roles (e.g. ['ADMIN'])
+ * Usage: app.get('/admin', requireAuth, requireRole(['ADMIN']), ...)
+ */
+export function requireRole(allowedRoles: string[]) {
+  return (req: AuthenticatedRequest, res: Response, next: NextFunction): Response | void => {
```

Biome (1.9.4)

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)